### PR TITLE
Cleanup in CardBrowser Part 1

### PR DESCRIPTION
**Pull request by oyeraghib**
## Purpose / Description
cleanup: replace TextUtils for checking empty or null 

## Learning (optional, can help others)
https://stackoverflow.com/a/59321544/9062752

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt [hunk 1]
```diff
@@ -909,9 +909,9 @@ open class CardBrowser :
             })
             // Fixes #6500 - keep the search consistent if coming back from note editor
             // Fixes #9010 - consistent search after drawer change calls invalidateOptionsMenu (mTempSearchQuery)
-            if (!TextUtils.isEmpty(mTempSearchQuery) || !TextUtils.isEmpty(mSearchTerms)) {
+            if (mTempSearchQuery!!.isNotEmpty() || mSearchTerms.isNotEmpty()) {
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:912]: `!TextUtils.isEmpty(mTempSearchQuery)` and `mTempSearchQuery!!.isNotEmpty()` are not equivalent, `mTempSearchQuery` is assigned null several lines above.

**Review by lukstbit — CHANGES_REQUESTED** [review]
You have failing tests.

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt (hunk 1 above)
Comment (BrayanDSO) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.kt:912]: Using `!isNullOrEmpty()` should do the trick

**Comment by oyeraghib** [comment]
Tests are failing here but passing on local

**Review by lukstbit — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
nice, thank you!

**Pull request merged**